### PR TITLE
[quant][graphmode] Move interpolate to general value ops

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38436 [quant] Histogram Observer support List[Tensor]
* #38435 [quant] Remove get_qparams in Observers
* #38434 [quant][graphmode] Add RemoveReduantDequantize pass
* #38359 [quant][graphmode][refactor] Factor out getFixedQParamOpFusionInfo
* #38358 [quant][graphmode][refactor] Factor out getInputTensorQParamOpFusionInfo
* #38278 [quant][graphmode] Support ops with fixed quantization parameters
* #38277 [quant][graphmode][refactor] insertDeQuantForAllUse
* #38276 [quant][graphmode][refactor] Refactor propagateQuantizationOps
* #38166 [quant][graphmode] Move leaky_relu to general value op map
* #38165 [quant][graphmode] Move elu to general value ops map
* #38164 [quant][graphmode] Move hardtanh to general value ops map
* #38163 [quant][graphmode] Move clamp to general value ops map
* **#38162 [quant][graphmode] Move interpolate to general value ops**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21559810](https://our.internmc.facebook.com/intern/diff/D21559810)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -1668,10 +1670,15 @@ def forward(self, x):
         # and for N general value op between conv we should have
         # N + 1 quantize_per_tensor between these ops
         m1 = convert_script(m, debug=True)
-        conv_op_quant = 4
         # NB: This Needs to be updated when we add more ops to test
-        general_value_op_quant = 14
-        FileCheck().check_count("aten::quantize_per_tensor(", conv_op_quant + general_value_op_quant + 1, exactly=True) \
+        # number of quantize_per_tensor op for type
+        num_quant_by_op_type = {'conv': 2, 'common': 1, 'interpolate': 3}
```
Comment (raghuramank100) [inline test/quantization/test_quantize_script.py:1675]: Can you explain why there are three quantize_per_tensor ops per interpolate op?

File: test/quantization/test_quantize_script.py [hunk 2]
```diff
@@ -1668,10 +1670,15 @@ def forward(self, x):
...
+        # number of ops for each type
+        num_op_by_op_type = {'conv': 2, 'common': 16, 'interpolate': 3}
```
Comment (raghuramank100) [inline test/quantization/test_quantize_script.py:1677]: It looks like there are 4 interpolate ops.

File: test/quantization/test_quantize_script.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1675]: there are some if statement in the code, and we have one quantize_per_tensor for each callsite of the aten ops inside if.

File: test/quantization/test_quantize_script.py (hunk 2 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1677]: probably need to rename this, 'interpolate' just means the ops that have three quantize_per_tensor..

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1677]: will rename this in later PR

Comment (supriyar) [inline test/quantization/test_quantize_script.py:1677]: What does common mean here?

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1677]: this is refactored in later PRs, 'common' just mean typical ops..

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1677]: typical op will introduce 1 quantize_per_tensor call in the IR.

**Comment by vkuzo** [comment]
stamping per @jerryzh168 's request

**Review by vkuzo — APPROVED** [review]

**Pull request closed without merging**